src/preferred-provider.js

```diff
@@ -61,20 +61,48 @@ function providerExists (uri) {
  *  given Web ID, extracted from Link rel header or profile body. If no
  *  provider URI was found, reject with an error.
  */
-function discoverProviderFor (webId) {
+function discoverProviderFor (webId, issuer) {
```

Comment:
So the method functionality has changed from discovering to validating? We probably need a method like `isValidProvider` then.

Reply:
Ok, I will update the code. 
I tried to create fix with a minimal changes in the code.